src/event.rs

```diff
@@ -372,7 +340,9 @@ where
 						hex_utils::to_string(&payment_hash.0),
 					);
 					self.channel_manager.fail_htlc_backwards(&payment_hash);
-					self.inbound_payments.lock().unwrap().remove(&payment_hash);
+					self.payment_store
+						.set_status(&payment_hash, PaymentStatus::Failed)
+						.expect("Failed to access payment store");
```

Comment:
is it expected behavior to crash here? just thinking from a user's perspective since you generally want to avoid _just_ crashing. or maybe that's okay for now?